cluster/swarm/cluster.go

```diff
@@ -496,59 +498,60 @@ func (c *Cluster) CreateNetwork(name string, request *types.NetworkCreate) (resp
 // CreateVolume creates a volume in the cluster
 func (c *Cluster) CreateVolume(request *types.VolumeCreateRequest) (*types.Volume, error) {
 	var (
-		wg     sync.WaitGroup
-		volume *types.Volume
-		err    error
 		parts  = strings.SplitN(request.Name, "/", 2)
-		node   = ""
+		config = &cluster.ContainerConfig{}
 	)
-
 	if request.Name == "" {
 		request.Name = stringid.GenerateRandomID()
 	} else if len(parts) == 2 {
-		node = parts[0]
+		config = cluster.BuildContainerConfig(containertypes.Config{Env: []string{"constraint:node==" + parts[0]}}, containertypes.HostConfig{}, networktypes.NetworkingConfig{})
 		request.Name = parts[1]
 	}
-	if node == "" {
-		c.RLock()
-		for _, e := range c.engines {
-			wg.Add(1)
-
-			go func(engine *cluster.Engine) {
-				defer wg.Done()
 
-				v, er := engine.CreateVolume(request)
-				if v != nil {
-					volume = v
-					err = nil
-				}
-				if er != nil && volume == nil {
-					err = er
-				}
-			}(e)
-		}
-		c.RUnlock()
+	nodes, err := c.scheduler.SelectNodesForContainer(c.listNodes(), config)
```

Comment:
For me if I provide a name like you have done, this call is failing saying that it can't find a node.